lib/engine.js

```diff
@@ -147,25 +147,19 @@ Engine.prototype._configureApp = function() {
     contracts: this.contracts,
     redis: this.redis
   });
-  const corsError = new Error('Not allowed by CORS');
   const corsOptions = {
     credentials: true,
     origin: function (origin, next) {
       const corsRegex = new RegExp(self._config.server.corsRegex);
       const originAllowed = function () {
         if (process.env.NODE_ENV !== 'production') {
-          log.warn('Insecure CORS configuration in use; non-production env detected');
           return true;
         }
 
         return corsRegex.test(origin);
       };
 
-      if (originAllowed()) {
-        return next(null, true);
-      }
-
-      next(corsError);
+      next(null, originAllowed());
```

Comment:
We always want the cors middleware to call `next` without an error. Calling `next` with a second argument of `true` sets the response `access-control-allow-origin` to the request's `origin`. We never want to prevent a request from being processed because there is not origin request header or if it doesn't match.

Reply:
the `cors` docs were misleading